Auto-grading with exact answers

dariusf opened the pull request:
Implements feature 1 in Shuqun's feature requests.

- On creating an assessment question, a UI for adding variable numbers of correct and wrong answers will be shown.
- When the question is graded, the saved answers for it are displayed. A grade will also be filled in automatically based on whether the submitted solution matches suggested answers.

dariusf commented on app/views/assessment/general_questions/_form.html.erb:
```diff
@@ -5,6 +5,34 @@
+        <%= f.input :auto_grading_type,
+            as: :radio_buttons,
+            label: 'Auto-grading type',
+            collection: [ ['None', 0], ['Exact match', 1], ['Keywords', 2]],
```
Rails 3 doesn't support ActiveRecord.enum. What's the preferred way to express this?

allenwq replied:
How about using this gem: https://github.com/lwe/simple_enum

lowjoel commented:
This kind of makes sense, so I think it's fine. Re specifics: @allenwq is better for v1.

allenwq commented on app/views/assessment/gradings/_single_question_form.html.erb:
```diff
@@ -45,6 +46,7 @@
         <hr>
         <div class="row-fluid">
           <div class="span10">
+            <%= render partial: 'auto_grading_exact_options', locals: {qn: qn} %>
```
I think you only want to render this partial for `general_questions` ?

allenwq commented:
@dariusf Can you add feature tests as well (to make sure that this feature works for submission/gradings)?  And some of the existing tests were break in this PR, please take a look.

dariusf commented:
Made use of `simple_enum` (had to change schema again to follow its conventions), fixed broken PDF export, added feature spec.

The spec only covers viewing saved auto-grading options. The interface for adding them involves JS without fallback.

allenwq commented:
@dariusf you forgot to push `Gemfile.lock`

dariusf commented:
Fixed simple_enum version, pushed Gemfile.lock. Also removed the placeholder text for keyword options (which aren't implemented as of this PR).
